app/components/atoms/CarouselCircle/CarouselCircle.styles.tsx

```diff
@@ -0,0 +1,27 @@
+import styled, {css} from 'styled-components';
+
+export interface CarouselCircleProps {
+  variant?: 'normal' | 'selected';
+}
+
+const variants = {
+  normal: css`
+    background-color: ${({theme}) => theme.colors.black};
+  `,
+  selected: css`
+    background-color: ${({theme}) => theme.colors.backgroundDarker};
+    border-width: 3px;
+    border-color: ${({theme}) => theme.colors.black};
+    border-style: solid;
+  `,
+};
+
+export const Bullet = styled.span<CarouselCircleProps>`
+  ${({variant}) => variants[variant || 'normal']}
+  display: block;
+
+  width: 10px;
+  height: 10px;
+
+  border-radius: 50%;
+`;
```

Comment:
nit: you can make your "normal" variant the default style:

```tsx

const activeStyles = css`
  background-color: ${({theme}) => theme.colors.backgroundDarker};
  border-width: 3px;
  border-color: ${({theme}) => theme.colors.black};
  border-style: solid;
`;

export const Bullet = styled.span<CarouselCircleProps>`
  background-color: ${({theme}) => theme.colors.black};
  border-radius: 50%;
  display: block;
  height: 10px;
  width: 10px;

  ${({ isActive }) => isActive && activeStyles}
`;

```

Idk if this snippet works, but it helps to get the idea. This way we don't need the variants mechanism.

Reply:
nice.

Reply:
Reuse this approach whenever you see this pattern: the component has a single variation based on one of its props.

This will work fine for inputs!